### PR TITLE
Add Trino 424 release notes

### LINKED ISSUE #18638: Release notes for 424

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 424 release. Fixes #18638 

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 11 Aug 2023

* #18640 ✅ rn ✅ docs
* #18570 ✅ rn ✅ docs
* #18592 ✅ rn ✅ docs
* #18559 ✅ rn ✅ docs
* #18641 ✅ rn ✅ docs
* #18646 ✅ rn ✅ docs
* #18648 ✅ rn ✅ docs
* #18624 ✅ rn ✅ docs
* #18471 ✅ rn ✅ docs
* #18639 ✅ rn ✅ docs
* #18657 ✅ rn ✅ docs

## 12 Aug 2023

* #18556 ✅ rn ✅ docs
* #18057 ✅ rn ✅ docs
* #18652 ✅ rn ✅ docs

## 14 Aug 2023

* #18663 ✅ rn ✅ docs
* #18493 ✅ rn ✅ docs
* #18588 ✅ rn ✅ docs
* #18669 ✅ rn ✅ docs
* #18672 ✅ rn ✅ docs
* #18554 ✅ rn ✅ docs
* #18558 ✅ rn ✅ docs
* #18675 ✅ rn ✅ docs
* #18658 ✅ rn ✅ docs
* #18118 ✅ rn ✅ docs

## 15 Aug 2023

* #18668 ✅ rn ✅ docs
* #18542 ✅ rn ✅ docs

## 16 Aug 2023

* #18689 ✅ rn ✅ docs
* #18684 ✅ rn ✅ docs
* #17510 ✅ rn ✅ docs
* #18701 ✅ rn ✅ docs
* #18703 ✅ rn ✅ docs
* #17509 ✅ rn ✅ docs
* #18707 ✅ rn ✅ docs
* #18686 ✅ rn ✅ docs
* #18692 ✅ rn ✅ docs
* #18696 ✅ rn ✅ docs
* #18681 ✅ rn ✅ docs
* #18680 ✅ rn ✅ docs
* #18679 ✅ rn ✅ docs
* #18677 ✅ rn ✅ docs
* #18676 ✅ rn ✅ docs
* #18678 ✅ rn ✅ docs
* #18700 ✅ rn ✅ docs

## 17 Aug 2023

* #18709 ✅ rn ✅ docs
* #18711 ✅ rn ✅ docs
* #18713 ✅ rn ✅ docs
* #18561 ✅ rn ✅ docs
* #18685 ✅ rn ✅ docs
* #18688 ✅ rn ✅ docs
* #18606 ✅ rn ✅ docs
* #18349 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-424.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+* Add support for `CASCADE` option in `DROP SCHEMA` statements. ({issue}`18689`)
+* Remove the legacy Parquet reader, along with the
+  `parquet.optimized-reader.enabled` and
+  `parquet.optimized-nested-reader.enabled` configuration properties. ({issue}`18639`)
+* Fix potential incorrect query results when a query involves a predicate on a
+  `timestamp with time zone` column. ({issue}`18588`)
+
+## Memory connector
+
+* Add support for `CASCADE` option in `DROP SCHEMA` statement. ({issue}`18668`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-424.md:42]:
```suggestion
* Add support for `CASCADE` option with `DROP SCHEMA`. ({issue}`18668`)
```

File: docs/src/main/sphinx/release/release-424.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## PostgreSQL connector
+
+* Add support for `CASCADE` option in `DROP SCHEMA` statements. ({issue}`18663`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-424.md:46]:
```suggestion
* Add support for `CASCADE` option with `DROP SCHEMA`. ({issue}`18663`)
```

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-424.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Release 424 (16 Aug 2023)
+
+## General
+
+* Improve query performance for large Trino clusters with many worker nodes. ({issue}`18542`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-424.md:5]: This PR doesn't improve query performance. It improves coordinator background overhead that's unrelated to query execution.

File: docs/src/main/sphinx/release/release-424.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+* Require the JVM default charset to be UTF-8. This can be set with the JVM
+  command line option `-Dfile.encoding=UTF-8`. ({issue}`18657`)
+
+## JDBC driver
+
+* Add the number of bytes written to HTTP to the query results response. ({issue}`18651`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-424.md:11]: What does "bytes written to HTTP" mean?

File: docs/src/main/sphinx/release/release-424.md (hunk 1 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-424.md:42]: The original phrasing sounds better.

Comment (mosabua) [reply docs/src/main/sphinx/release/release-424.md:42]: Okay.. we just did similar changes recently. I like the original one more as well.

File: docs/src/main/sphinx/release/release-424.md (hunk 3 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-424.md:5]: Does it need a release note, then?

**Comment by nineinchnick** [comment]
```
% ./report.sh -s 423 -t 424
Changes per version:
version = 423
 status = removed
  names = hive:parquet.optimized-nested-reader.enabled, hive:parquet.optimized-reader.enabled

version = 424
 status = new
  names = :experimental.thread-per-driver-scheduler-enabled, :fault-tolerant-execution-runtime-adaptive-partitioning-enabled, :fault-tolerant-execution-runtime-adaptive-partitioning-max-task-size, :fault-tolerant-execution-runtime-adaptive-partitioning-partition-count
```

I guess we don't want to mention the new experimental flag, but what about the FTE flags?

**Pull request merged**